### ISSUE TITLE
Support for exposing /metrics and /healthz?

### ISSUE THREAD

**Issue by dudicoco** [zalando-incubator/kube-metrics-adapter#98]
Hi,

Does kube-metrics-adapter exposes its own metrics with a prometheus client? Does it expose an endpoint for health checks using kubernetes probes?

Thanks

**Comment by mikkeloscar** [comment]
No, we currently don't provide either of those endpoints. What sort of metrics would you want to get from a prometheus endpoint?

**Comment by dudicoco** [comment]
Hi @mikkeloscar, I would like to get error counts so I can set up alerts them.
I would also like a health endpoint so I can configure liveness probe checks to restart the container if it's unhealthy.
Here is an example of an implementation in the External-DNS controller:
https://github.com/kubernetes-sigs/external-dns/pull/132#issuecomment-292490308

**Comment by mikkeloscar** [comment]
I guess it can make sense to report errors, e.g. if a collector is not working you could figure it out. maybe just a bit hard to distinguish between user configuration errors and actual errors in some cases.

PRs for adding such a metric endpoint would be welcome.

Regarding the health endpoint I'm not sure under which condition it would make sense to kill the container. Unlike external-dns, this is an active HTTP server so it definitely shouldn't have a separate listener for `/healthz`, but it could have one on the main listener and be used for readieness, though it's pretty much ready right away, anyway :)

**Comment by dudicoco** [comment]
Thanks for the detailed explanation @mikkeloscar.

So if I set up a probe with https scheme to target the main listener i'm good to go?
Do you only recommend setting a readiness probe and not a liveness probe?

**Comment by mikkeloscar** [comment]
We didn't run with either in production for a year or so now, and we didn't have issues related to this thus far, so I can't recommend on either as I don't know what the benefit would be :)
